core/azure-core/src/main/java/com/azure/core/http/policy/TokenCredentialPolicy.java

```diff
@@ -26,7 +28,7 @@ public TokenCredentialPolicy(TokenCredential credential) {
 
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
-        return credential.getTokenAsync(context.httpRequest().url().toString())
+        return credential.getTokenAsync(Arrays.asList(context.httpRequest().url().toString() + ".default"))
```

Comment:
@schaabs I know this is probably not the correct way of finding out the scope from a request. But this has worked in many services when we had resource instead of scopes.

I guess I'm still not sure about the benefits of scopes. To me, it still feels like an over-engineered abstraction over resource, through which ugliness (adding `.default` and splitting by `.`) is spilled everywhere.

Reply:
If you have some docs about how to use scopes correctly I would really appreciate it.